src/components/TrackTile/hooks/useTrackerValues.ts

```diff
@@ -93,9 +93,17 @@ export const useTrackerValues = (
     if (!loading && !_.isEqual(prevDateRange, dateRange)) {
       setLoading(true);
 
-      svc
-        .fetchTrackerValues(valuesContext, dateRange)
-        .then((fetchedTrackerValues) => {
+      Promise.resolve()
+        .then(async () => {
+          if (valuesContext.shouldUseOntology) {
+            // NOTE: The service uses ontology for values roll-up.  We don't need
+            // to anything with the result here; we just need the service to fetch.
+            await svc.fetchOntology(valuesContext.codeBelow);
+          }
```

Comment:
Great question- it's a major "this will work for now".  I think there's a bit of a smell, IMO, in `useAxiosTrackTileService`.  To put words to it, it feels like that service/impl is doing too much.  This PR (obviously) doesn't address that at all, in fact it adds to it just a bit.  But I bring it up because you can imagine this: the axios service is injected to the highest level Provider in track-tile world.  I didn't want to "always fetch ontologies", to avoid extra requests for customers that won't use/need ontological roll-up.  So how do we utilize the appConfig, and its configuration of advancedScreens, and inject that into the service?  After giving up on useAxiosTrackTileService itself, I tried `useTrackers` and found similar difficulties.  Being directly related to the values, I found that this was a sensible way to easily utilize props passed through from a given project's appConfig. 

Very open to thoughts, concerns, and esp. suggestions.

Reply:
I agree the current state of the hooks and especially the axios track tile service are less than ideal given the current use case. I think you're this works for now. Hopefully we get a chance to refactor the service at some point to use react-query and the queryClient caching.